lib/trento/activity_logging/activity_catalog.ex

```diff
@@ -107,7 +107,11 @@ defmodule Trento.ActivityLog.ActivityCatalog do
             false
         end)
         |> Enum.map(&Module.concat/1)
-        |> Enum.filter(&(not function_exported?(&1, :legacy, 0)))
+        |> Enum.filter(fn module ->
+          Code.ensure_loaded?(module)
```

Comment:
I wonder what is the expected/desired behavior here in case of successful or unsuccessful response ? Additionally, what are the scenarios at run time when this will return an error response ?

Reply:
We're calling `Code.ensure_loaded?(module)` to make sure that the following [`function_exported?`](https://hexdocs.pm/elixir/Kernel.html#function_exported?/3) works as expected.

> what is the expected/desired behavior here in case of successful or unsuccessful response
>
Unless something very wrong with beam, I wouldn't expect `Code.ensure_loaded?(module)` to fail in this context as `module` is always an actual module (probably not loaded yet) that we get from `:application.get_key(:trento, :modules)`.

> what are the scenarios at run time when this will return an error response
>
None that I am aware of, unless something very wrong with beam is going on.

---

Alternatively to this we could just always have a `legacy?/0` function in events returning a boolean: like here e9be79658f0d124346c7216c017ba5ce74aac182